corehq/apps/ip_access/models.py

```diff
@@ -25,27 +28,24 @@ def __repr__(self):
         )
 
     def is_allowed(self, ip_address):
-        should_check_country = True
-        if not self.country_allowlist:
-            if not settings.MAXMIND_LICENSE_KEY:
-                should_check_country = False
-            elif ip_address not in self.ip_allowlist:
-                return False
-        return (
-            ip_address not in self.ip_denylist
-            and (
-                ip_address in self.ip_allowlist
-                or (should_check_country and is_in_country(ip_address, self.country_allowlist))
-            )
-        )
+        if ip_address in self.ip_denylist or (self.country_allowlist and not settings.MAXMIND_LICENSE_KEY):
+            return False
```

Comment:
I think you can drop the `or..` part of this conditional.  As-is, this means it won't call `is_in_country`, and the 403 will never be raised.